spring-integration-jms/src/main/java/org/springframework/integration/jms/JmsSendingMessageHandler.java

```diff
@@ -70,6 +70,7 @@ public void setDestinationExpression(Expression destinationExpression) {
 		Assert.isTrue(this.destination == null && this.destinationName == null,
 				"The 'destination', 'destinationName', and 'destinationExpression' properties are mutually exclusive.");
 		this.destinationExpressionProcessor = new ExpressionEvaluatingMessageProcessor<Object>(destinationExpression);
+		setPrimaryExpression(destinationExpression);
```

Comment:
`HttpRequestExecutingMessageHandler.uriExpression`, `MongoDbStoringMessageHandler.collectionNameExpression`.
Although, yes, I agree that we can do that on demand.

`AggregatingMessageHandler` is a `AbstractCorrelatingMessageHandler`, therefore `correlationStrategy`'s `expression` is promiry. 
Just  logical flow :smile: .